### PR TITLE
Integrate backend for searching courses

### PR DESCRIPTION
**Pull request by haiyingweng**
## Overview

Integrate backend for searching courses

## Changes Made

- make request to backend for search result with input query
- display search result
- convert data to camelCase

## Test Coverage

visual

## Screenshots (delete if not applicable)

<details>

  <summary>Recording</summary>

https://user-images.githubusercontent.com/48968041/113660051-4fd8a100-9671-11eb-8c19-b5eb2c7ae105.mov

  

</details>

### CODE REVIEW

File: src/utils/requests.ts [hunk 1]
```diff
@@ -1,44 +1,57 @@
...
 
 export const setAuthHeader = (token: string | null) => {
-  const header = `Bearer ${token ? token : localStorage.getItem('accessToken')}`
-  axios.defaults.headers.common['Authorization'] = header
+    const header = `Bearer ${token ? token : localStorage.getItem('accessToken')}`
+    axios.defaults.headers.common['Authorization'] = header
 }
 
-export const validateToken = async () => {}
+export const validateToken = async () => { }
```
Comment (danielVebman) [inline src/utils/requests.ts:40]: The space looks weird to me. If it's auto formatted like this then whatever.

File: src/utils/requests.ts [hunk 2]
```diff
@@ -1,44 +1,57 @@
...
     const res = await api.get(url, headers, params)
     const { success, data } = res.data
     if (success) return data
     throw Error(data)
 }
-  
-const post = async (url: string, headers?: any, body?: any) => {
-    const res = await api.post(url, headers, body)
+
+const post = async (url: string, body?: any, headers?: any,) => {
```
Comment (danielVebman) [inline src/utils/requests.ts:28]: Why the trailing comma?

File: src/components/Search/SearchView.tsx [hunk 3]
```diff
@@ -38,7 +54,11 @@ class SearchView extends React.Component {
           {this.searchIcon(18, 18)}
           <input className="search-input"
             value={this.state.searchText}
-            onChange={(event) => this.setState({ searchText: event.target.value })} />
+            onChange={(event) => {
```
Comment (danielVebman) [inline src/components/Search/SearchView.tsx:57]:
```suggestion
            onChange={ (event) => {
```
Space before `(event)`

File: src/components/Search/SearchView.tsx [hunk 4]
```diff
@@ -2,33 +2,49 @@ import React from 'react'
...
+        <circle cx="6.5" cy="6.5" r="5.5" stroke="#9AA0A6" strokeWidth="2" />
         <rect x="9.74316" y="11.1411" width="2" height="8.75923" rx="1" transform="rotate(-44.5128 9.74316 11.1411)" fill="#9AA0A6" />
       </svg>
     )
   }
 
+  getCourses = async (query: string) => {
+    if (this.state.searchText.length > 2) {
+      try {
+        const searchResult: CourseQuery = await searchCourses(query)
```
Comment (danielVebman) [inline src/components/Search/SearchView.tsx:32]: Do you need the type annotation?

File: src/components/Search/SearchView.tsx [hunk 5]
```diff
@@ -38,7 +54,11 @@ class SearchView extends React.Component {
           {this.searchIcon(18, 18)}
           <input className="search-input"
             value={this.state.searchText}
-            onChange={(event) => this.setState({ searchText: event.target.value })} />
+            onChange={(event) => {
+              this.setState({ searchText: event.target.value })
```
Comment (danielVebman) [inline src/components/Search/SearchView.tsx:58]: Is this the right type? What about `courses`?

File: src/components/Search/SearchView.tsx [hunk 6]
```diff
@@ -2,33 +2,49 @@ import React from 'react'
...
 
+import { searchCourses } from '../../utils/requests';
+
+import { Course, CourseQuery } from '../../types';
+
+type SearchViewState = {
+  searchText: string
+  courses: Course[]
+}
+
```
Comment (danielVebman) [inline src/components/Search/SearchView.tsx:13]: Is there a point in explicitly defining this type? I'm honestly not sure what's better style.

**Review by danielVebman — APPROVED** [review]
Looks good. Just a couple small comments.

File: src/utils/requests.ts (hunk 1 above)
Comment (haiyingweng) [reply src/utils/requests.ts:40]: yea it's the formatter

File: src/utils/requests.ts (hunk 2 above)
Comment (haiyingweng) [reply src/utils/requests.ts:28]: not intended

File: src/components/Search/SearchView.tsx (hunk 3 above)
Comment (haiyingweng) [reply src/components/Search/SearchView.tsx:57]: formatter removes the space

File: src/components/Search/SearchView.tsx (hunk 6 above)
Comment (haiyingweng) [reply src/components/Search/SearchView.tsx:13]: needed it to define type for courses in state

File: src/components/Search/SearchView.tsx (hunk 5 above)
Comment (haiyingweng) [reply src/components/Search/SearchView.tsx:58]: do you mean what happens to `courses`?  `setState()` will keep `courses` same as what was there

**Review by YKo20010 — APPROVED** [review]
🚀 🚀 🤩 lgtm

**Pull request merged**